docs/style-guide.md

```diff
@@ -0,0 +1,564 @@
+}
+```
+
+## Functions
+
+### Return early from functions
+
+To avoid deep nesting of if-statements, always return a function's value as early
+as possible.
+
+*Right:*
+
+```js
+function isPercentage(val) {
+  if (val < 0) {
+    return false;
+  }
+
+  if (val > 100) {
+    return false;
+  }
+
+  return true;
+}
+```
+
+*Wrong:*
+
+```js
+function isPercentage(val) {
+  if (val >= 0) {
+    if (val < 100) {
+      return true;
+    } else {
+      return false;
+    }
+  } else {
+    return false;
+  }
+}
+```
+
+Or for this particular example it may also be fine to shorten things even
+further:
+
+```js
+function isPercentage(val) {
+  const isInRange = (val >= 0 && val <= 100);
+  return isInRange;
+}
+```
+
+### Name your closures
+
+Feel free to give your closures a name. It shows that you care about them, and
+will produce better stack traces, heap and cpu profiles.
+
+*Right:*
+
+```js
+req.on('end', function onEnd() {
+  console.log('winning');
+});
+```
+
+*Wrong:*
+
+```js
+req.on('end', function() {
+  console.log('losing');
+});
+```
+
+### No nested closures
+
+Use closures, but don't nest them. Otherwise your code will become a mess.
+
+*Right:*
+
+```js
+setTimeout(function() {
+  client.connect(afterConnect);
+}, 1000);
+
+function afterConnect() {
+  console.log('winning');
+}
+```
+
+*Wrong:*
+
+```js
+setTimeout(function() {
+  client.connect(function() {
+    console.log('losing');
+  });
+}, 1000);
+```
+
+
+### Method chaining
+
+One method per line should be used if you want to chain methods.
+
+You should also indent these methods so it's easier to tell they are part of the same chain.
+
+*Right:*
+
+```js
+User
+  .findOne({ name: 'foo' })
+  .populate('bar')
+  .exec(function(err, user) {
+    return true;
+  });
+```
+
+*Wrong:*
+
+```js
+User
+.findOne({ name: 'foo' })
+.populate('bar')
+.exec(function(err, user) {
+  return true;
+});
+
+User.findOne({ name: 'foo' })
+  .populate('bar')
+  .exec(function(err, user) {
+    return true;
+  });
+
+User.findOne({ name: 'foo' }).populate('bar')
+.exec(function(err, user) {
+  return true;
+});
+
+User.findOne({ name: 'foo' }).populate('bar')
+  .exec(function(err, user) {
+    return true;
+  });
+```
+
+## Comments
+
+### Use slashes for comments
+
+Use slashes for both single line and multi line comments. Try to write
+comments that explain higher level mechanisms or clarify difficult
+segments of your code. Don't use comments to restate trivial things.
+
+*Right:*
+
+```js
+// 'ID_SOMETHING=VALUE' -> ['ID_SOMETHING=VALUE', 'SOMETHING', 'VALUE']
+const matches = item.match(/ID_([^\n]+)=([^\n]+)/));
+
+// This function has a nasty side effect where a failure to increment a
+// redis counter used for statistics will cause an exception. This needs
+// to be fixed in a later iteration.
+function loadUser(id, cb) {
+  // ...
+}
+
+const isSessionValid = (session.expires < Date.now());
+if (isSessionValid) {
+  // ...
+}
+```
+
+*Wrong:*
+
+```js
+// Execute a regex
```

Comment:
Looks like `wrong` section of examples uses valid comments syntax

Reply:
It describes the point about: "Don't use comments to restate trivial things".